src/components/JfListMakerComponent.spec.js

```diff
@@ -1,50 +1,389 @@
+const INPUT_TEXT = `input.input-text`;
+const INPUT_DROPDOWN = 'jf-ui-select-stub'//This is -stub since the child is stubbed
+
+//Events
+const EVENT_ON_ADD_VALUE = 'on-add-value';
+const EVENT_ON_AFTER_ADD_VALUE = 'on-after-add-value';
+const EVENT_INPUT = 'input';
+const EVENT_ON_AFTER_DELETE_VALUE = 'on-after-delete-value';
+
+//Messages
+const MSG_MUST_INPUT_VALUE = `Must input value`;
+const MSG_VALUE_ALREADY_EXISTS = `Value already exists`
+
 describe('JfListMakerComponent', () => {
-    it('renders a form', () => {
-        const wrapper = shallowMount(JfListMakerComponent, {
-            propsData: {},
-            stubs: ['jf-field']
+    describe("display", function() {
+        const wrapper = mountComponent();
+        it('should render a form', () => {
+            expect(wrapper.contains('form')).toBe(true);
         });
-        expect(wrapper.contains('form')).toBe(true);
-    });
 
+        it('should render a + button to add new elements to the list', () => {
+            expect(wrapper.contains(BUTTON_ADD)).toBe(true);
+        })
+        it('The + button is disabled by default', () => {
+            expect(
+                wrapper.find(BUTTON_ADD).attributes('disabled')
+            ).toBe('disabled')
+        })
+        it('The + button should be enabled when any text is typed in', () => {
+            wrapper.find(INPUT_TEXT).setValue('a')
+            expect(
+                wrapper.find(BUTTON_ADD).attributes('disabled')
+            ).not.toBe('disabled')
+        })
+        it('If ng-disabled prop is passed, the + button is disabled even if content is typed in', () => {
+            const wrapper = mountComponent({
+                "ng-disabled": true,
+            });
+            wrapper.find(INPUT_TEXT).setValue("a");
+            expect(wrapper.find(BUTTON_ADD).attributes("disabled")).toBe("disabled")
+        })
 
-    it('renders a + button to add new elements to the list', () => {
-        const wrapper = shallowMount(JfListMakerComponent, {
-            propsData: {},
-            stubs:["jf-field"]
-        });
-        expect(wrapper.contains('i.icon.icon-new')).toBe(true);
-    })
-    it('The + button is disabled by default', () => {
-        const wrapper = shallowMount(JfListMakerComponent, {
-            propsData: {},
-            stubs:["jf-field"]
-        });
-        expect(
-            wrapper.find('.icon.icon-new').attributes('disabled')
-        ).toBe('disabled')
+        it('should display passed in values in the list', function () {
+            const wrapper = mountComponent({
+                value: ['a'],
+            })
+            //text
+            expect(wrapper.find(SELECTOR_A_LIST_ROW).text()).toEqual("a")
+            //delete button
+            expect(
+                wrapper
+                    .contains(
+                        BUTTON_DELETE
+                    )
+            ).toBe(true)
+        })
+
+        it('should display new values in the list', function () {
```

Comment:
When you do mainly html tree checks, you can use jest's snapshot testing. So you can mount and snapshot the component, very useful and 'cheap' to write IMHO.
https://vue-test-utils.vuejs.org/guides/#snapshot-testing

Reply:
Thank you. Will add this for jf-billboard tests